identities/models.py

```diff
@@ -44,3 +47,59 @@ class Identity(models.Model):
 
     def __str__(self):
         return str(self.id)
+
+    def remove_details(self, user):
+        updated_details = {}
+        for attribute, value in self.details.items():
+            updated_details[attribute] = "removed"
+        self.details = updated_details
+        self.updated_by = user
+        self.save()
+
+
+class OptOut(models.Model):
```

Comment:
Still missing the `request_type` which should default to `unsubscribe` and also support `forget`. Remove details should only be triggered if it's `forget`.